xDrip/Services/CloudSync/Codables/CTreatment.swift

```diff
@@ -0,0 +1,258 @@
+               let intensityString = TrainingIntensity(rawValue: intensityValue)?.paramValue() {
+                exerciseIntensity = intensityString
+            } else {
+                exerciseIntensity = nil
+            }
+        }
+        
+        eventType = treatmentType.rawValue.capitalized
+        
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSSZ"
+        sysTime = dateFormatter.string(from: entry.date ?? Date())
+        
+        dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss'Z'"
+        dateFormatter.timeZone = TimeZone(secondsFromGMT: 0)
+        createdAt = dateFormatter.string(from: entry.date ?? Date())
+        
+        utcOffset = nil
+        timestamp = Int((entry.date ?? Date()).timeIntervalSince1970)
+        enteredBy = "xdrip iOS"
+        insulinInjections = "[]"
+        
+        if treatmentType == .carbs, let carbEntry = entry as? CarbEntry {
+            foodType = carbEntry.foodType
+        } else {
+            foodType = nil
+        }
+    }
+    
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        
+        treatmentID = try? container.decode(String.self, forKey: .treatmentID)
+        carbs = try? container.decode(Double.self, forKey: .carbs)
+        eventType = try? container.decode(String.self, forKey: .eventType)
+        insulin = try? container.decode(Double.self, forKey: .insulin)
+        uuid = try? container.decode(String.self, forKey: .uuid)
+        createdAt = try? container.decode(String.self, forKey: .createdAt)
+        sysTime = try? container.decode(String.self, forKey: .sysTime)
+        utcOffset = try? container.decode(Int.self, forKey: .utcOffset)
+        timestamp = try? container.decode(Int.self, forKey: .timestamp)
+        enteredBy = try? container.decode(String.self, forKey: .enteredBy)
+        insulinInjections = try? container.decode(String.self, forKey: .insulinInjections)
+        foodType = try? container.decode(String.self, forKey: .foodType)
+        duration = try? container.decode(Double.self, forKey: .duration)
+        exerciseIntensity = try? container.decode(String.self, forKey: .exerciseIntensity)
+        
+        switch eventType?.lowercased() {
+        case TreatmentType.carbs.rawValue.lowercased():
+            type = .carbs
+        case TreatmentType.bolus.rawValue.lowercased():
+            type = .bolus
+        case TreatmentType.basal.rawValue.lowercased():
+            type = .basal
+        case TreatmentType.training.rawValue.lowercased():
+            type = .training
+        default:
+            type = .carbs
+        }
+    }
+    
+    static func getIDFromUUID(uuid: String?) -> String {
+        guard let externalID = uuid else { return "" }
+        if externalID.count <= 24 {
+            return externalID
+        } else {
+            let str = externalID.replacingOccurrences(of: "-", with: "")
```

Comment:
Add a comment explaining this logic. I get why you replace "-" with "", but why are you doing substring?